src/test/java/com/emc/object/s3/S3JerseyClientTest.java

```diff
@@ -2942,4 +2943,101 @@ protected void assertAclEquals(AccessControlList acl1, AccessControlList acl2) {
+            add(new Tag().withKey("k5").withValue("v5"));
+            add(new Tag().withKey("k6").withValue("v6"));
+            add(new Tag().withKey("k7").withValue("v7"));
+            add(new Tag().withKey("k8").withValue("v8"));
+            add(new Tag().withKey("k9").withValue("v9"));
+            add(new Tag().withKey("k10").withValue("v10"));
+        }};
+        List<Tag> tagsExceeded = new ArrayList<Tag>(){{addAll(tags); add(new Tag().withKey("k11").withValue("v11"));}};
+        PutObjectTaggingRequest putObjectTaggingRequestSingleTag = new PutObjectTaggingRequest(bucketName, okey)
+                        .withTagging(new Tagging().withTagSet(new TagSet().withTags(tag))),
+                putObjectTaggingRequestMultipleTags = new PutObjectTaggingRequest(bucketName, okey)
+                        .withTagging(new Tagging().withTagSet(new TagSet().withTags(tags))),
+                putObjectTaggingRequestExceededTags = new PutObjectTaggingRequest(bucketName, okey)
+                        .withTagging(new Tagging().withTagSet(new TagSet().withTags(tagsExceeded)));
+        // get different versions of the existing object
+        client.setBucketVersioning(bucketName, new VersioningConfiguration().withStatus(VersioningConfiguration.Status.Enabled));
+        client.putObject(bucketName, okey, "Hello VersionID0 !", "text/plain");
+        String versionId0 = client.listVersions(bucketName, null).getVersions().get(0).getVersionId();
+        client.putObject(bucketName, okey, "Hello VersionID1 !", "text/plain");
+        String versionId1 = client.listVersions(bucketName, null).getVersions().get(0).getVersionId();
+        client.putObject(bucketName, okey, "Hello VersionID2 !", "text/plain");
+        String versionId2 = client.listVersions(bucketName, null).getVersions().get(0).getVersionId();
+
+        // should be able to update tags for a particular version of an object
+        client.putObjectTagging(putObjectTaggingRequestSingleTag.withVersionId(versionId0));
+        Assert.assertEquals(1, client.getObject(new GetObjectRequest(bucketName, okey).withVersionId(versionId0), String.class).getObjectMetadata().getTaggingCount());
+        client.putObjectTagging(putObjectTaggingRequestMultipleTags.withVersionId(versionId1));
+        Assert.assertEquals(10, client.getObject(new GetObjectRequest(bucketName, okey).withVersionId(versionId1), String.class).getObjectMetadata().getTaggingCount());
+        // should not be able to add more than 10 tags per object
+        try {
+            client.putObjectTagging(putObjectTaggingRequestExceededTags.withVersionId(versionId2));
+        } catch (S3Exception e) {
+            Assert.assertEquals(400, e.getHttpCode());
+            Assert.assertEquals("invalid content length", e.getErrorCode());
+        }
+
+    }
+
+    @Test
+    public void testDeleteObjectTagging() {
+        // set up env
+        String bucketName = getTestBucket(), okey = "test-object-tagging";
```

Comment:
Maybe you could test 3 operations deleteObjectTagging(), getObjectTagging() deleteObjectTagging() together in one test case, and some other test cases for tagging with Put/Get Object, init MPU, Put Object Copy request?